src/Core/Framework/Plugin/Util/AssetService.php

```diff
@@ -49,7 +50,12 @@ public function copyAssetsFromBundle(string $bundleName): void
         }
 
         $targetDirectory = $this->getTargetDirectory($bundle);
-        $this->filesystem->deleteDir($targetDirectory);
+
+        try {
+            $this->filesystem->deleteDir($targetDirectory);
```

Comment:
implement own filesystem adapter which is extending from the GCP and catch that error there. On the application level, we should not catch specific filesystem exceptions

Reply:
That's fair. I will create an MR for `superbalist/flysystem-google-cloud-storage` so it catches the exception there which makes Shopware work as intended :+1:

Reply:
Upstream patch created: https://github.com/Superbalist/flysystem-google-cloud-storage/pull/122

Reply:
Yes I will! this MR is closed for now, which is fine, will bump it when merged upstream!